state-validate-util/src/main/java/org/sunbird/statevalidateutil/RecordProcessor.java

```diff
@@ -0,0 +1,206 @@
+    public void startProcessingStateValidationAndFlags() throws IOException {
+        String custodianOrgId = getSystemSettingCustodianOrgId();
+        List<User> usersList = getRecordsToBeProcessed();
+        usersList
+                .stream()
+                .forEach(
+                        userObject -> {
+                            Map<String, String> compositeKeysMap = getCompositeKeysMap(userObject);
+                            try {
+                                startTracingRecord(userObject.getUserId());
+                                boolean stateValidated = isStateUSer(userObject.getUserId(), custodianOrgId);
+                                Map<String, Object> userBooleanMap = new HashMap<>();
+                                userBooleanMap.put(DbColumnConstants.stateValidated, stateValidated);
+                                userBooleanMap.put(DbColumnConstants.emailVerified, userObject.getEmailVerified());
+                                userBooleanMap.put(DbColumnConstants.phoneVerified, userObject.getPhoneVerified());
+                                userObject.setFlagsValue(calcFieldsValue(userBooleanMap));
+                                performSequentialOperationOnRecord(userObject, compositeKeysMap, stateValidated);
+                            } catch (Exception e) {
+                                logExceptionOnProcessingRecord(compositeKeysMap);
+                            } finally {
+                                endTracingRecord(userObject.getUserId());
+                            }
+                        });
+
+        connection.closeConnection();
+        closeWriterConnection();
+    }
+
+    private int calcFieldsValue(Map<String, Object> userBooleanMap) {
+        int userFlagValue = 0;
+        Set<Map.Entry<String, Object>> mapEntry = userBooleanMap.entrySet();
+        for(Map.Entry<String, Object> entry: mapEntry) {
+            if(StringUtils.isNotEmpty(entry.getKey()) && (Boolean) entry.getValue()) {
+                userFlagValue += boolFlagValue(entry.getKey());
+            }
+        }
+        return userFlagValue;
+    }
+
+    public int boolFlagValue(String userFlagType) {
+        int decimalValue = 0;
+        if(userFlagType.equals(UserBoolFlagEnum.PHONE_VERIFIED.getUserFlagType())) {
+            decimalValue = UserBoolFlagEnum.PHONE_VERIFIED.getUserFlagValue();
+        } else if (userFlagType.equals(UserBoolFlagEnum.EMAIL_VERIFIED.getUserFlagType())) {
+            decimalValue = UserBoolFlagEnum.EMAIL_VERIFIED.getUserFlagValue();
+        } else if (userFlagType.equals(UserBoolFlagEnum.STATE_VALIDATED.getUserFlagType())) {
+            decimalValue = UserBoolFlagEnum.STATE_VALIDATED.getUserFlagValue();
+        }
+        return decimalValue;
+    }
+
+    private boolean isStateUSer(String userId, String custodianOrgId) {
+        String orgId = null;
+        ResultSet resultSet = connection.getRecords("select organisationid from user_org where userid='"+userId+"'");
```

Comment:
Inside user we already having rootOrgId , so no need to make extra call

Reply:
yes, done the change